src/java.net.http/share/classes/jdk/internal/net/http/Http1Request.java

```diff
@@ -81,15 +81,22 @@ class Http1Request {
 
     private void logHeaders(String completeHeaders) {
         if (Log.headers()) {
-            //StringBuilder sb = new StringBuilder(256);
-            //sb.append("REQUEST HEADERS:\n");
-            //Log.dumpHeaders(sb, "    ", systemHeaders);
-            //Log.dumpHeaders(sb, "    ", userHeaders);
-            //Log.logHeaders(sb.toString());
-
-            String s = completeHeaders.replaceAll("\r\n", "\n");
-            if (s.endsWith("\n\n")) s = s.substring(0, s.length() - 2);
-            Log.logHeaders("REQUEST HEADERS:\n{0}\n", s);
+            StringBuilder sb = new StringBuilder(completeHeaders.length());
+            sb.append("REQUEST HEADERS:\n");
+            boolean[] firstLine = {true};
+            completeHeaders.lines().forEach(line -> {
+                // First line contains `GET /foo HTTP/1.1`.
+                // Convert it to look like other `Log.logHeaders()` outputs.
+                if (firstLine[0]) {
+                    int protocolFieldPrefixIndex = line.lastIndexOf(' ');
+                    assert protocolFieldPrefixIndex > 0;
+                    sb.append("  ").append(line, 0, protocolFieldPrefixIndex).append('\n');
```

Comment:
I'd rather print the whole line here - including HTTP/1.1.
Nice to have it indented differently though.

Reply:
Implemented in 7b958db6f4.